### ISSUE TITLE
Encoding issues with many popular sites

### ISSUE THREAD

**Issue by pirate** [webrecorder/pywb#434]
## Describe the bug
When using `wayback` in both `--proxy` and `--proxy-record` mode, many sites seem to display gibberish, possibly because of an incorrect encoding being forced somewhere.

Working sites:
 - https://example.com
 - https://reddit.com
 - https://github.com
 - and many more...

Broken sites:
 - https://google.com
 - https://facebook.com
 - https://www.cloudflare.com
 - and many more...

I'm not sure what makes some sites work and not others.

## Steps to reproduce the bug
1. Use Python 3.7.2 and either pywb v2.1.0 or pywb-2.2.0.dev0 (latest `develop` branch)
2. Using an empty `config.yaml`, run:
```bash
wb-manager init demo
wayback --proxy-record --proxy demo
```
3. Open a page in a browser:
```bash
google-chrome --proxy-server=http://localhost:8080 --ignore-certificate-errors --disable-web-security https://cloudflare.com
```

## Expected behavior
A readable site is recorded and replayed instead of ��������

## Screenshots
![image](https://user-images.githubusercontent.com/511499/51600020-03a94800-1ecf-11e9-98bd-af26531c360b.png)

## Environment
- OS: macOS 10.14
- Browser: Google Chrome 72.0.3626.64 beta
- Version: Python 3.7.2 and pywb v2.1.0 [e.g. 22]

**Comment by N0taN3rd** [comment]
Have you tried using a non-empty config file and ensured that pywb auto-generated a CA ([docs](https://pywb.readthedocs.io/en/latest/manual/configuring.html#https-proxy-and-pywb-certificate-authority)) and or the dev branch?

I was not able to reproduce the issue using the config below when running from the dev branch using 3.7.2 and linux.
 
```yml
collections:
    all: $all
    live: $live

# Settings for each collection
use_js_obj_proxy: true

# Memento support, enable
enable_memento: true

# Replay content in an iframe
framed_replay: true
```

**Comment by pirate** [comment]
It correctly autogenerated a CA, and the issue persists with the config you gave, except now the frame is tiny in the upper left corner of the screen:

<img width="625" alt="screen shot 2019-01-24 at 4 33 11 pm" src="https://user-images.githubusercontent.com/511499/51709956-d0ada400-1ff5-11e9-9694-a77c09dd9d2d.png">

**Comment by ikreymer** [comment]
Thanks for sending the zip, unfortunately, can't repro the issue that you're seeing. The WARC in demo does contains valid HTML, but contains only the html for cloudflare, not any of the stylesheets (presumably since they didn't load in the browser).
I've tried both with latest Chrome and Chrome Canary, everything records and replays correctly.

Perhaps there is some other service running on your system that could interfere? It might be worth trying a different port than 8080, just in case. You can specify it with `-p` flag to change the port.
The HTML is Brotli encoded, but it should be getting decoded for rewriting, and pywb auto-decodes if the browser doesn't support Brotli anyway.

Other things to try: try upgrading the gevent library with `pip install -U gevent` in case there is an issue with that. The logs look a bit sparse, there should be more log entries made then that if everything is working.

**Comment by pirate** [comment]
Very strange, maybe it's a macOS thing? Or an encoding problem with my shell? What's your `LC_ALL` and `LANG`?

Is there a more verbose log mode I can use?  I'll try upgrading gevent and running it in a fresh macOS user account and then a linux VM.

**Comment by ikreymer** [comment]
I'm on a mac as well.. `LC_ALL` is unset and `LANG=en_US.UTF-8`

**Comment by ikreymer** [comment]
This happens in url rewriting/non-proxy mode as well? What happens if you load `http://localhost:8080/demo/record/https://cloudflare.com/` either in the browser, or `curl -L http://localhost:8080/live/mp_/https://cloudflare.com/` ?

**Comment by pirate** [comment]
Two new findings: it only happens in Chrome, and I don't think it's an encoding issue.  I tried decoding the page in a number of different encodings with Chrome, and none of them looked like real text.  Could it be that it's not ungzipping the page for some reason in Chrome, and trying to show the compressed bytes as UTF-8 anyway?

While it appears to only happen in Chrome, but it happens reliably in all versions of Chrome on my machine (tested with fresh sessions & no extensions on Chrome Stable, Beta, and Canary).

It works fine in Safari & curl for both replaying and viewing, and in both proxy and rewriting modes.  I would love to just use Safari, but unfortunately the primary reason I'm setting up pywb is to use it as a proxy recorder for scripted Chrome headless web archiving.

**Comment by pirate** [comment]
Blocked by: https://github.com/python-hyper/brotlipy/issues/146

**Comment by ikreymer** [comment]
The core issue was brotli not being installed properly. But, now pywb will not record in brotli if it can't decode it. Fixed in 2.2 release.

**Comment by pirate** [comment]
Thanks @ikreymer !
